apis/database/v1alpha2/rdsinstance_types.go

```diff
@@ -66,52 +66,691 @@ func (v *DBSubnetGroupNameReferencerForRDSInstance) Assign(res resource.CanRefer
+	// Amazon Aurora
+	// Not applicable. The encryption for DB instances is managed by the DB cluster.
+	// For more information, see CreateDBCluster.
+	// Default: false
+	// +immutable
+	// +optional
+	StorageEncrypted *bool `json:"storageEncrypted,omitempty"`
+
+	// StorageType specifies the storage type to be associated with the DB instance.
+	// Valid values: standard | gp2 | io1
+	// If you specify io1, you must also include a value for the IOPS parameter.
+	// Default: io1 if the IOPS parameter is specified, otherwise standard
+	// +optional
+	StorageType *string `json:"storageType,omitempty"`
+
+	// Tags. For more information, see Tagging Amazon RDS Resources (http://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/USER_Tagging.html)
+	// in the Amazon RDS User Guide.
+	// +immutable
+	// +optional
+	Tags []Tag `json:"tags,omitempty"`
+
+	// TODO(muvaf): get this password as input when we have a way of supplying
+	// sensitive information as input from the user.
+	// TDECredentialArn is the ARN from the key store with which to associate the instance for TDE encryption.
+	// +optional
+	// TDECredentialArn *string `json:"tdeCredentialArn,omitempty"`
+	// TDECredentialPassword is the password for the given ARN from the key store in order to access the
+	// device.
+	// +optional
+	//TDECredentialPassword *string `json:"tdeCredentialPassword,omitempty"`
+
+	// Timezone of the DB instance. The time zone parameter is currently supported
+	// only by Microsoft SQL Server (http://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/CHAP_SQLServer.html#SQLServer.Concepts.General.TimeZone).
+	// +immutable
+	// +optional
+	Timezone *string `json:"timezone,omitempty"`
+
+	// VPCSecurityGroupIDs is a list of EC2 VPC security groups to associate with this DB instance.
+	// Amazon Aurora
+	// Not applicable. The associated list of EC2 VPC security groups is managed
+	// by the DB cluster. For more information, see CreateDBCluster.
+	// Default: The default EC2 VPC security group for the DB subnet group's VPC.
+	// +optional
+	VPCSecurityGroupIDs []string `json:"vpcSecurityGroupIds,omitempty"`
+
+	// A value that specifies that the DB instance class of the DB instance uses
+	// its default processor features.
+	UseDefaultProcessorFeatures *bool `json:"useDefaultProcessorFeatures,omitempty"`
 }
 
 // An RDSInstanceSpec defines the desired state of an RDSInstance.
 type RDSInstanceSpec struct {
 	runtimev1alpha1.ResourceSpec `json:",inline"`
-	RDSInstanceParameters        `json:",inline"`
+	ForProvider                  RDSInstanceParameters `json:"forProvider,omitempty"`
+
+	// DBSubnetGroupNameRef references to a DBSubnetGroup to retrieve its name
+	// +immutable
+	// +optional
+	DBSubnetGroupNameRef *DBSubnetGroupNameReferencerForRDSInstance `json:"dbsubnetGroupNameRef,omitempty" resource:"attributereferencer"`
+
+	// VPCSecurityGroupIDRefs references to a list of SecurityGroups to retrieve a list of securityGroupIDs
+	// +immutable
```

Comment:
is +immutable feature actually supported by Kubebuilder?
https://book.kubebuilder.io/reference/markers/crd-validation.html

Reply:
Hopefully in the future, but we sit tight. https://github.com/crossplaneio/crossplane/blob/master/design/one-pager-managed-resource-api-design.md#immutable-properties